File: DataDriven/Laravel/app/Http/Controllers/AuthorController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use App\Models\Author;
use App\Models\Post;
use Illuminate\Support\Facades\Concurrency;

class AuthorController extends Controller
{
    public function getPosts(Request $request)
    {
        try {
            $emails = $request->input('emails');
            $authors = Author::whereIn('email', $emails)->get();

            if ($authors->isEmpty()) {
                return response()->json(['result' => 'No authors found.'], 200);
            }

            $authorIDs = $authors->pluck('id');
            $postsByAuthor = Post::whereIn('id', function ($query) use ($authorIDs) {
                $query->select('post')
                    ->from('author_post')
                    ->whereIn('author', $authorIDs);
            })->get();

            if ($postsByAuthor->isEmpty()) {
                return response()->json(['result' => 'No posts found.'], 200);
            } else {
                return response()->json(['result' => $postsByAuthor], 200);
            }
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }
    public function newAuth(Request $request)
    {
        set_time_limit(0);

        $commentsData = $request->input('cats', []);
        if (!is_array($commentsData) || count($commentsData) < 1) {
            return response()->json(['error' => 'Invalid request data.'], 422);
        }

        try {
            $chunkedArray = array_chunk($commentsData, 100);
            foreach ($chunkedArray as $index => $chunk) {
                set_time_limit(0);

                $userCount = Concurrency::run(function () use ($chunk) {
                    Author::insert($chunk);
                });
            }
            return response()->json(['success' => true], 200);
        } catch (\Exception $e) {
            return response()->json(['error' => $e->getMessage()], 500);
        }
    }
}

File: DataDriven/Phoenix/app/lib/app_web/controllers/author_controller.ex
defmodule AppWeb.AuthorController do
  use AppWeb, :controller

  alias App.Repo
  alias App.Blog.Author
  alias App.Blog.Post
  import Ecto.Query, only: [from: 2]

  def get_posts(conn, %{"emails" => emails}) do
    if !is_list(emails) or length(emails) < 1 do
      json(conn, %{error: "Invalid request data."})
      |> halt()
    end

    authors = Repo.all(from a in Author, where: a.email in ^emails)

    if length(authors) < 1 do
      json(conn, %{result: "No authors found."})
    else
      author_ids = Enum.map(authors, & &1.id)

      posts_by_author =
        Repo.all(
          from p in Post,
            join: pa in "post_authors",
            on: pa.post_id == p.id,
            where: pa.author_id in ^author_ids,
            select: %{post: p}
        )

      if length(posts_by_author) < 1 do
        json(conn, %{result: "No posts found."})
      else
        json(conn, %{result: posts_by_author})
      end
    end
  rescue
    e ->
      json(conn, %{error: "#{e}"})
      IO.inspect(e, label: "Error")
  end
end
